### DISCUSSION TITLE
Spruce support

### DISCUSSION THREAD

**Discussion by HugoDF** [alpine-collective/alpinejs-devtools#175] [Remnant]
Maybe one for @ryangjchandler, what would it take for the devtools to have a "stores" tab? 

If I remember correctly, Spruce has a methods (outside of Alpine) for:
- accessing stores: https://docs.ryangjchandler.co.uk/spruce/stores#accessing-a-store-from-javascript
- set data in store: https://docs.ryangjchandler.co.uk/spruce/stores#overwriting-a-store
- watching a store: https://docs.ryangjchandler.co.uk/spruce/watchers

("stores" > "spruce" so that we're forward compatible if stores make it into v3)

The tricky bit might be to be able to access the same `Spruce` as the user, if we can do `window.Spruce` we should be alright though.

**Comment by ryangjchandler** [comment]
Yeah, Spruce is on the window so you can access all of the methods you mentioned. 

It wouldn't be too hard, other than listening for global state changes. Watchers would work but might not be the most performing. 

I could add a hook into Spruce for that though, would mean a minimum version requirement is needed.

**Reply by HugoDF** [reply]
How about listing all the stores? That feasible with the current API?

**Reply by ryangjchandler** [reply]
> How about listing all the stores? That feasible with the current API?

Yeah, `Object.keys(Spruce.stores)` should do it.

**Reply by ryangjchandler** [reply]
I'm kind of hoping Alpine v3 doesn't have native stores, purely because Spruce has a wider range of functionality than Alpine would (based on what I've seen). Either way, we can support both perhaps..

**Reply by HugoDF** [reply]
In that case I guess there'll be a use case for Spruce as a V2/V3 abstraction for stores.

Ok cool, I might convert this into an issue since it all seems doable.
